script-library/common-debian.sh

```diff
@@ -254,7 +254,7 @@ __bash_prompt() {
     local userpart='`export XIT=$? \
         && [ ! -z "${GITHUB_USER}" ] && echo -n "\[\033[0;32m\]@${GITHUB_USER} " || echo -n "\[\033[0;32m\]\u " \
         && [ "$XIT" -ne "0" ] && echo -n "\[\033[1;31m\]➜" || echo -n "\[\033[0m\]➜"`'
-    local gitbranch='`export BRANCH=$(git describe --contains --all HEAD 2>/dev/null); \
+    local gitbranch='`export BRANCH=$(git rev-parse --abbrev-ref HEAD 2>/dev/null || git describe --contains --all HEAD 2>/dev/null); \
```

Comment:
We need to check whether the output equals "HEAD" and then fallback. I'm not sure exactly how to do that in a one-liner, but something like this seems to work locally for me:

```
BRANCH=$(git rev-parse --abbrev-ref HEAD 2>/dev/null)
if [ "${BRANCH}" == "HEAD" ]; then
    BRANCH=$(git describe --contains --all HEAD 2>/dev/null);
fi
```

Reply:
A quick way to test is to checkout a branch, and then run `git checkout HEAD~1`. When running this script, the outcome should be `{branch}~1` instead of `HEAD`.

Reply:
One liner for that would be
```
export BRANCH=$(git rev-parse --abbrev-ref HEAD 2>/dev/null);  test "${BRANCH}" = "HEAD" || export BRANCH=$(git describe --contains --all HEAD 2>/dev/null);
```
... but that's less readable.  I'll update.

Reply:
Ok - updated! Thanks for checking that.